testsuite/ui/navigation/step_processor.py

```diff
@@ -0,0 +1,84 @@
+import inspect
```

Comment:
I don't think we need additional file for this class as it is only used only in the Navigator and is not general enough to be used on its own